neuro_extras/main.py

```diff
@@ -484,108 +441,13 @@ def data_cp(
-            /tmp/refdisk_manifest.json
-    neuro-extras data cp s3://sra-pub-sars-cov2/sra-src/SRR9967744/ storage:
-    """
-    run_async(_data_cp(source, destination, extract, volume, env))
-
-
-@image.command("build")
-@click.option("-f", "--file", default="Dockerfile")
-@click.option("--build-arg", multiple=True)
-@click.option(
-    "-v",
-    "--volume",
-    metavar="MOUNT",
-    multiple=True,
-    help=(
-        "Mounts directory from vault into container. "
-        "Use multiple options to mount more than one volume. "
-        "Use --volume=ALL to mount all accessible storage directories."
-    ),
-)
-@click.option(
-    "-e",
-    "--env",
-    metavar="VAR=VAL",
-    multiple=True,
-    help=(
-        "Set environment variable in container "
-        "Use multiple options to define more than one variable"
-    ),
-)
-@click.argument("path")
-@click.argument("image_uri")
-def image_build(
-    file: str,
-    build_arg: Sequence[str],
-    path: str,
-    image_uri: str,
-    volume: Sequence[str],
-    env: Sequence[str],
-) -> None:
-    run_async(_build_image(file, path, image_uri, build_arg, volume, env))
-
-
-@image.command("copy")
-@click.argument("source")
-@click.argument("destination")
-def image_copy(source: str, destination: str) -> None:
-    run_async(_copy_image(source, destination))
-
-
-@main.command("cp")
-@click.argument("source")
-@click.argument("destination")
-def cluster_copy(source: str, destination: str) -> None:
-    run_async(_copy_storage(source, destination))
-
-
-async def _copy_storage(source: str, destination: str) -> None:
-    src_uri = uri_from_cli(source, "", "")
-    src_cluster = src_uri.host
-    src_path = src_uri.parts[2:]
-
-    dst_uri = uri_from_cli(destination, "", "")
-    dst_cluster = dst_uri.host
-    dst_path = dst_uri.parts[2:]
-
-    assert src_cluster
-    assert dst_cluster
-    async with neuro_api.get() as client:
-        await client.config.switch_cluster(dst_cluster)
-        await client.storage.mkdir(URL("storage:"), parents=True, exist_ok=True)
-    await _run_copy_container(src_cluster, "/".join(src_path), "/".join(dst_path))
-
-
-async def _run_copy_container(src_cluster: str, src_path: str, dst_path: str) -> None:
-    args = [
-        "neuro",
-        "run",
-        "-s",
-        "cpu-small",
-        "--pass-config",
-        "-v",
-        "storage:://storage",
-        "-e",
-        f"NEURO_CLUSTER={src_cluster}",
-        NEURO_EXTRAS_IMAGE,
-        f"neuro cp --progress -r -u -T storage:{src_path} /storage/{dst_path}",
-    ]
-    cmd = " ".join(args)
-    print(f"Executing '{cmd}'")
-    subprocess = await asyncio.create_subprocess_shell(cmd)
-    returncode = await subprocess.wait()
-    if returncode != 0:
-        raise Exception("Unable to copy storage")
+    if extract and compress:
+        raise click.ClickException("Extract and compress can't be used together")
```

Comment:
why it does not work? overcomplicates the code?
(just curious, definitely repacking cases are rare)

Reply:
Right, this seems like an unnecessary code path. We can support it of course, but I imagine in most case this combination will be an error.